### PR TITLE
[Scala] new way to Init

### PR DESCRIPTION
**Pull request by Ldpe2G**

### CODE REVIEW

File: scala-package/core/src/main/scala/ml/dmlc/mxnet/Initializer.scala [hunk 1]
```diff
@@ -33,6 +33,10 @@ abstract class Initializer {
 
     if (name.startsWith("upsampling")) {
       initBilinear(name, arr)
+    } else if (name.startsWith("stn_loc") && name.endsWith("weight")) {
```
Comment (piiswrong) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/Initializer.scala]: we don't do this anymore.
name matching is a bad idea. use Variable(..., init=xx) instead

Comment (Ldpe2G) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/Initializer.scala]: @piiswrong implementation of converting initializer object to json string and back to object in Scala is a bit complex. I am still thinking a way to implement the same functionality of python's initializer in Scala. So for now the Scalapkg still use the legacy way to init.

Comment (javelinjs) [inline scala-package/core/src/main/scala/ml/dmlc/mxnet/Initializer.scala]: @Ldpe2G how about reflection?

**Comment by Ldpe2G** [comment]
@javelinjs @piiswrong I figure out a way to implement the new initialization.
@javelinjs would you help to review ?

**Comment by Ldpe2G** [comment]
@javelinjs @piiswrong I find out this implementation is wrong.

**Pull request closed without merging**